src/identifiers/migrations/0009_deduplicate_identifiers_20220527.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2022-03-15 20:06
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def deduplicate_identifiers(apps, schema_editor):
+    Identifier = apps.get_model("identifiers", "Identifier")
+    Journal = apps.get_model("journal", "Journal")
+    journals = Journal.objects.all()
+    for journal in journals:
+        identifiers_in_journal = Identifier.objects.filter(article__journal=journal)
+        for id_type in ['doi', 'uri', 'pubid']:
+            identifiers_of_type = identifiers_in_journal.filter(id_type=id_type)
+            for doi_string in set(identifiers_of_type.values_list('identifier', flat=True)):
+                to_keep_pk = identifiers_of_type.filter(identifier=doi_string).values_list('id', flat=True)[0]
+                duplicate_pks = identifiers_of_type.filter(identifier=doi_string).values_list('id', flat=True)[1:]
+                duplicate_identifiers = [identifier for identifier in Identifier.objects.filter(pk__in=duplicate_pks)]
+                print(id_type, to_keep_pk, doi_string)
+                if duplicate_identifiers:
+                    print('\n\n\n')
+                    print('To keep:')
+                    print(id_type, to_keep_pk, doi_string)
+                    print('Duplicates:')
+                    for dup in duplicate_identifiers:
+                        print(id_type, dup.pk, dup.identifier)
+                    print('\n\n\n')
+                Identifier.objects.filter(pk__in=duplicate_pks).delete()
```

Comment:
I don't think we need all these printing.
Also, we are keeping the first identifier for each duplicate... makes me wonder if this is a good approach, since we won't be able to detect which of these are mistaken. I would advocate for deleting duplicate IDs for a given article but refrain from doing this for duplicates across an entire journal in case we wipe a correct identifier while preserving the wrong one